### PR TITLE
upgrade to v2 canary, simplify vite config

### PR DESCRIPTION
**Pull request by alvrs**
<details><summary>Edit: error is fixed</summary>
<p>
TODO: client build fails with this error. Need to investigate further.
CleanShot 2023-03-28 at 23 29 29@2x
</p>
</details> 

This PR upgrades v2sandbox to the latest canary release of mud, which always corresponds to the latest mud main (https://github.com/latticexyz/mud)

### CODE REVIEW

**Comment by alvrs** [comment]
Gonna fix the vercel issue with a different approach in a followup.

File: packages/contracts/package.json [hunk 1]
```diff
@@ -7,16 +7,16 @@
...
-    "deploy:testnet": "yarn prepare && mud deploy-v2 --profile=lattice-testnet",
+    "deploy": "yarn prepare && yarn mud deploy-v2",
+    "deploy:testnet": "yarn prepare && yarn mud deploy-v2 --profile=lattice-testnet",
     "lint": "yarn prettier && yarn solhint",
-    "tablegen": "mud tablegen",
-    "worldgen": "mud worldgen",
-    "worldtypes": "rimraf types && typechain --target=ethers-v5 out/world/IWorld.sol/IWorld.json",
+    "tablegen": "yarn mud tablegen",
+    "worldgen": "yarn mud worldgen",
+    "worldtypes": "rimraf types && (typechain --target=ethers-v5 out/IWorld.sol/IWorld.json || typechain --target=ethers-v5 out/world/IWorld.sol/IWorld.json)",
```
Comment (holic) [inline packages/contracts/package.json:19]: 😭

File: packages/contracts/package.json [hunk 2]
```diff
@@ -7,16 +7,16 @@
...
     "init": "yarn tablegen && yarn build && yarn worldgen && yarn worldtypes",
     "prepare": "yarn tablegen && yarn worldgen && yarn worldtypes",
+    "devnode": "yarn mud devnode",
     "build": "forge clean && forge build",
-    "devnode": "mud devnode",
     "test": "echo todo",
     "dev": "yarn deploy",
-    "deploy": "yarn prepare && mud deploy-v2 --clean",
-    "deploy:testnet": "yarn prepare && mud deploy-v2 --profile=lattice-testnet",
+    "deploy": "yarn prepare && yarn mud deploy-v2",
```
Comment (holic) [inline packages/contracts/package.json:14]: did you find that you needed `yarn` ahead of `mud` for it to work? my understanding of scripts is they run contextually inside the package/with the dependency binaries available - is that not true?

**Review by holic — APPROVED** [review]

File: packages/contracts/package.json (hunk 2 above)
Comment (alvrs) [reply packages/contracts/package.json:14]: I was under the impression it didn't work without yarn but let me double check that!

**Pull request merged**